tests/mtouch/InlinerTest.cs

```diff
@@ -0,0 +1,157 @@
+		static bool DisplayCandidates = false;
+
+		protected abstract string Assembly { get; }
+
+		AssemblyDefinition assembly;
+
+		protected virtual AssemblyDefinition AssemblyDefinition {
+			get {
+				if (assembly == null)
+					assembly = AssemblyDefinition.ReadAssembly (Assembly);
+				return assembly;
+			}
+		}
+
+		protected string ListMethods (HashSet<string> h)
+		{
+			string result = Path.GetFileName (Assembly);
+			if (h.Count == 0)
+				return $" {result}: success";
+			return result + "\n" + h.Aggregate ((arg1, arg2) => arg1 + "\n" + arg2);
+		}
+
+		bool IsCandidateForInlining (MethodDefinition m)
+		{
+			// candidates must be methods
+			if (m.IsConstructor)
+				return false;
+			// must have a body (IL)
+			if (!m.HasBody)
+				return false;
+			// must be static or not virtual (can't be overrriden)
+			if (!m.IsStatic && m.IsVirtual)
+				return false;
+			// the body must not have exception handlers or variables
+			var b = m.Body;
+			return !(b.HasExceptionHandlers || b.HasVariables || b.InitLocals);
+		}
+
+		protected void NoParameterReturnOnlyConstant (Code code, HashSet<string> list)
+		{
+			if (DisplayCandidates)
+				Console.WriteLine ($"### NoParameterReturnOnlyConstant {code}: {Path.GetFileName (Assembly)}");
+
+			var ad = AssemblyDefinition.ReadAssembly (Assembly);
+			foreach (var t in ad.MainModule.Types) {
+				if (!t.HasMethods)
+					continue;
+				foreach (var m in t.Methods) {
+					if (!IsCandidateForInlining (m))
+						continue;
+					if (m.HasParameters)
+						continue;
+					var b = m.Body;
+					if (b.Instructions.Count != 2)
+						continue;
+					var ins = b.Instructions [0];
+					if (code == ins.OpCode.Code) {
+						var s = m.ToString ();
+						list.Remove (s);
+						if (DisplayCandidates)
+							Console.WriteLine ($"* `{s}`");
+					}
+				}
+			}
+		}
+
+		protected void NoParameterNoReturnNoCode (HashSet<string> list)
+		{
+			if (DisplayCandidates)
+				Console.WriteLine ($"### ReturnOnly: {Path.GetFileName (Assembly)}");
+
+			foreach (var t in AssemblyDefinition.MainModule.Types) {
+				if (!t.HasMethods)
+					continue;
+				foreach (var m in t.Methods) {
+					if (!IsCandidateForInlining (m))
+						continue;
+					if (m.HasParameters)
+						continue;
+					if (!m.ReturnType.Is ("System", "Void"))
+						continue;
+					var b = m.Body;
+					if (b.Instructions.Count == 1) {
+						var s = m.ToString ();
+						list.Remove (s);
+						if (DisplayCandidates)
+							Console.WriteLine ($"* `{s}`");
+					}
+				}
+			}
+		}
+	}
+
+	[TestFixture]
+	public class MscorlibInlinerTest : InlinerTest {
+
+		protected override string Assembly {
+			get { return Path.Combine (Configuration.MonoTouchRootDirectory, "lib", "mono", "Xamarin.iOS", "mscorlib.dll"); }
+		}
+
+		/// <summary>
+		/// We look for candidates, without parameters, that only do `return true;`.
+		/// Such methods can be inlined by replacing the `call` with a `ldc.i4.1` instruction
```

Comment:
Is this true for instance calls as well, or will replacing the `call` with a `ldc.*` leave the instance on the IL stack?

You might also have to consider tail calls (not a problem for the methods in this PR, but it might be a problem for a more general scenario).